rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -112,8 +145,22 @@ void SqliteStorage::write(std::shared_ptr<const rosbag2_storage::SerializedBagMe
             "' has not been created yet! Call 'create_topic' first.");
   }
 
+#ifdef EN_TRANSACTION
+  if (!active_transaction_) {
+    activate_transaction();
+  }
+  // To work with a batched insert within a trasaction
+  no_of_inserts++;
+#endif
+
   write_statement_->bind(message->time_stamp, topic_entry->second, message->serialized_data);
   write_statement_->execute_and_reset();
```

Comment:
with the transactions open, is this command staying the same?
Are the actual transactions orthogonal to the individual commits to the database?

Reply:
@sriramster, do you mind answering to this?